tests/purs/.gitattributes

```diff
@@ -0,0 +1 @@
+*.out.js -text
```

Comment:
This, for anyone who's curious, is my solution to the `tests/purs/optimize` tests failing on Windows. Git has some sort of heuristic for determining whether a file is text—I suspect it's based on the extension—and if it believes a file is text, by default (at least on the GitHub runners?) it will normalize the newlines in that file to the operating system's default when the file is checked out, and map them back on commit. (Obviously, this breaks tests if it happens to golden test files.) `.gitattributes` files can be used to tell Git, among other things, that certain files are not text and therefore shouldn't have this behavior.

There are other approaches—we could use `git config` to disable newline conversion globally on GitHub runners, for example, but that wouldn't fix things for developers on Windows. I'm refreshing some rather old memories about all this now so I don't know if this is the best possible solution, but it seems pretty good.

Reply:
We're doing the same thing for the golden layout tests here: https://github.com/purescript/purescript/blob/fcf0acbcf11ad113820d32ce9ee82365a6015c9c/tests/purs/layout/.gitattributes#L1

(Maybe consider also adding -merge here)

Reply:
Oh thanks, that makes me feel better about this. Must not be extension-based then; probably it's sniffing out non-printable characters or something.

I don't think -merge is needed here; I think these files will merge pretty cleanly.

Reply:
The `-merge` is to make sure Git doesn't automatically merge when pulling. Golden files always need to be accepted wholesale, trying to "merge" them by combining lines from multiple PR's is basically always wrong.

Reply:
Isn't the important question whether the changes made to the files have good locality? If branch A changes the way imports are encoded in JS, and branch B changes the way that TCO loops get compiled, both of those branches change the compiled JS output but in different parts of the file, and Git's merge algorithm will probably successfully produce the golden file which should result from the combination of the features, right?